src/main/java/com/google/sps/data/Tab.java

```diff
@@ -0,0 +1,50 @@
+package com.google.sps.data;
+
+import java.util.Objects;
+
+public final class Tab{
+    private final String name;
+    private final String platform;
+    private final long id;
+
+    public Tab(String name, String platform, long id){
+        this.name = name;
+        this.platform = platform;
+        this.id = id; 
+    }
+
+    public String getTabName(){
+        return name; 
+    }
+
+    public String getPlatform(){
+        return platform; 
+    }
+
+    public long getID(){
+        return id; 
+    }
+
+    @Override
+    public boolean equals(Object o) {   
+        if (!(o instanceof Tab)) { 
+            return false; 
+        } 
+        Tab that = (Tab) o;
+        return that.getPlatform().equals(this.platform)
+            && that.getTabName().equals(this.name);
```

Comment:
why is id not used?

Reply:
we hadn't been adding ids to any of the equals functions because we only wanted to test for equality of the content of the objects as opposed to the id related to their entity

Reply:
ok makes sense